Framework/include/QualityControl/runnerUtils.h

```diff
@@ -74,87 +74,44 @@ inline bool hasChecks(std::string configSource)
-{ // Determine period
-  std::string periodName;
-  periodName = services.get<framework::RawDeviceService>().device()->fConfig->GetProperty<std::string>("periodName", "unspecified");
-  ILOG(Debug, Devel) << "Got this property periodName from RawDeviceService: '" << periodName << "'" << ENDM;
-  periodName = periodName != "unspecified" /* found it in service */ ? periodName : fallbackPeriodName;
-  ILOG(Info, Devel) << "Period Name returned by computePeriodName : " << periodName << ENDM;
-  return periodName;
-}
-
-inline std::string computePassName(const std::string& fallbackPassName = "")
-{
-  ILOG(Debug, Devel) << "Pass Name returned by computePassName : " << fallbackPassName << ENDM;
-  return fallbackPassName;
-}
-
-inline std::string computeProvenance(const std::string& fallbackProvenance = "")
-{
-  ILOG(Debug, Devel) << "Provenance returned by computeProvenance : " << fallbackProvenance << ENDM;
-  return fallbackProvenance;
+  result = result > 0 /* found it in service */ ? result : fallbackNumber;
+  ILOG(Info, Devel) << name << " returned by computeRunNumber (default) : " << result << ENDM;
+  return result;
 }
 
 inline Activity computeActivity(framework::ServiceRegistryRef services, const Activity& fallbackActivity)
 {
-  int runNumber = computeRunNumber(services, fallbackActivity.mId);
-  Activity activity{
+  auto runNumber = computeActivityField<int>(services, "runNumber", fallbackActivity.mId);
+  auto runType = computeActivityField<int>(services, "runType", fallbackActivity.mType);
+  auto run_start_time_ms = computeActivityField<unsigned long>(services, "runStartTimeMs", fallbackActivity.mValidity.getMin());
+  auto run_stop_time_ms = computeActivityField<unsigned long>(services, "runEndTimeMs", fallbackActivity.mValidity.getMax());
+  auto partitionName = services.get<framework::RawDeviceService>().device()->fConfig->GetProperty<std::string>("environment_id", fallbackActivity.mPartitionName);
+  auto periodName = services.get<framework::RawDeviceService>().device()->fConfig->GetProperty<std::string>("lhcPeriod", "unspecified");
+  auto fillNumber = computeActivityField<int>(services, "fillInfoFillNumber", fallbackActivity.mFillNumber);
+  auto beam_type = services.get<framework::RawDeviceService>().device()->fConfig->GetProperty<std::string>("fillInfoBeamType", fallbackActivity.mBeamType);
+
+  Activity activity(
     runNumber,
-    computeRunType(services, fallbackActivity.mType),
-    computePeriodName(services, fallbackActivity.mPeriodName),
-    computePassName(fallbackActivity.mPassName),
-    computeProvenance(fallbackActivity.mProvenance),
-    fallbackActivity.mValidity
-  };
-  // if available we overwrite with the info from logbook.
-  Bookkeeping::getInstance().populateActivity(activity, runNumber);
+    runType,
+    fallbackActivity.mPeriodName,
+    fallbackActivity.mPassName,
+    fallbackActivity.mProvenance,
```

Comment:
why fallback activity instead of taking the computed values if they are available?